packages/wallets/src/features/cashier/modules/DepositCrypto/DepositCrypto.tsx

```diff
@@ -8,12 +12,27 @@ import DepositCryptoTryFiatOnRamp from './components/DepositCryptoTryFiatOnRamp/
 import './DepositCrypto.scss';
 
 const DepositCrypto = () => {
+    const { isSuccess: isAuthorizeSuccess } = useAuthorize();
+    const { data: depositCryptoAddress, error, isLoading, mutate: mutateDepositCrypto } = useDepositCryptoAddress();
+
+    useEffect(() => {
+        if (isAuthorizeSuccess) {
+            mutateDepositCrypto();
+        }
+    }, [isAuthorizeSuccess, mutateDepositCrypto]);
+
+    if (isLoading) return <Loader />;
+
+    if (isServerError(error?.error)) {
+        return <DepositErrorScreen error={error?.error} />;
+    }
+
     return (
         <div className='wallets-deposit-crypto'>
             <div className='wallets-deposit-crypto__side-pane' />
             <div className='wallets-deposit-crypto__main-content'>
                 <DepositCryptoCurrencyDetails />
-                <DepositCryptoAddress />
+                <DepositCryptoAddress depositCryptoAddress={depositCryptoAddress} isLoading={isLoading} />
```

Comment:
@lubega-deriv 
believe that isLoading here is redundant since we are checking it on line 24.